File: step_definitions/Secure Login with Credentials and Social Media.java
package com.cucumber.steps;

import com.page_objects.LoginPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class SecureLoginSteps extends LoginPage {

    @Given("the user navigates to the login page")
    public void theUserNavigatesToTheLoginPage() {
        navigateToLoginPage();
    }

    @When("the user enters valid username {string} and password {string}")
    public void theUserEntersValidUsernameAndPassword(String username, String password) {
        enterUserName(username);
        enterPassword(password);
    }

    @When("clicks on the 'Login' button")
    public void clicksOnTheLoginButton() {
        clickLoginButton();
    }

    @Then("the user is successfully logged in and redirected to the dashboard")
    public void theUserIsSuccessfullyLoggedInAndRedirectedToTheDashboard() {
        verifyDashboardRedirection();
    }

    @Given("the user is logged in")
    public void theUserIsLoggedIn() {
        verifyUserLoggedIn();
    }

    @When("the user logs out of the account")
    public void theUserLogsOutOfTheAccount() {
        clickLogoutButton();
    }

    @Then("the user is redirected to the login page")
    public void theUserIsRedirectedToTheLoginPage() {
        verifyLoginPageRedirection();
    }

    @When("the user selects 'Login with Facebook'")
    public void theUserSelectsLoginWithFacebook() {
        clickLoginWithFacebook();
    }

    @Then("the Facebook login page is displayed")
    public void theFacebookLoginPageIsDisplayed() {
        verifyFacebookLoginPageDisplayed();
    }

    @When("the user enters Facebook credentials and logs in")
    public void theUserEntersFacebookCredentialsAndLogsIn() {
        enterFacebookCredentialsAndLogin();
    }

    @Then("the user is successfully logged in through Facebook and redirected to the dashboard")
    public void theUserIsSuccessfullyLoggedInThroughFacebookAndRedirectedToTheDashboard() {
        verifyDashboardRedirection();
    }

    @Given("the user is logged in through Facebook")
    public void theUserIsLoggedInThroughFacebook() {
        verifyUserLoggedInThroughFacebook();
    }

    @When("the user selects 'Login with Google'")
    public void theUserSelectsLoginWithGoogle() {
        clickLoginWithGoogle();
    }

    @Then("the Google login page is displayed")
    public void theGoogleLoginPageIsDisplayed() {
        verifyGoogleLoginPageDisplayed();
    }

    @When("the user enters Google credentials and logs in")
    public void theUserEntersGoogleCredentialsAndLogsIn() {
        enterGoogleCredentialsAndLogin();
    }

    @Then("the user is successfully logged in through Google and redirected to the dashboard")
    public void theUserIsSuccessfullyLoggedInThroughGoogleAndRedirectedToTheDashboard() {
        verifyDashboardRedirection();
    }

    @When("the user attempts to log in")
    public void theUserAttemptsToLogIn() {
        attemptLogin();
    }

    @Then("CAPTCHA is displayed and required for login")
    public void captchaIsDisplayedAndRequiredForLogin() {
        verifyCaptchaDisplayed();
    }

    @Then("no security warnings or alerts are displayed")
    public void noSecurityWarningsOrAlertsAreDisplayed() {
        verifyNoSecurityWarnings();
    }

    @Given("the user navigates to the login page from a new device")
    public void theUserNavigatesToTheLoginPageFromANewDevice() {
        navigateToLoginPageFromNewDevice();
    }

    @Then("verification via email or phone is required")
    public void verificationViaEmailOrPhoneIsRequired() {
        verifyEmailOrPhoneVerificationRequired();
    }

    @When("the user is inactive for a period")
    public void theUserIsInactiveForAPeriod() {
        simulateUserInactivity();
    }

    @Then("the session times out and requires re-login")
    public void theSessionTimesOutAndRequiresReLogin() {
        verifySessionTimeout();
    }

    @Then("all login pages and redirects are secured with HTTPS")
    public void allLoginPagesAndRedirectsAreSecuredWithHTTPS() {
        verifyHttpsRedirection();
    }

    @Given("the system is under maintenance")
    public void theSystemIsUnderMaintenance() {
        simulateSystemMaintenance();
    }

    @Then("the user receives a maintenance notification and cannot log in")
    public void theUserReceivesAMaintenanceNotificationAndCannotLogIn() {
        verifyMaintenanceNotification();
    }
}

File: step_definitions/Secure Login with Credentials and Social Media.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import LoginPage from "../pages/loginPage";

Given("the user navigates to the login page", () => {
  LoginPage.visit();
});

When("the user enters valid username {string} and password {string}", (username, password) => {
  LoginPage.enterUsername(username);
  LoginPage.enterPassword(password);
});

When("clicks on the 'Login' button", () => {
  LoginPage.clickLoginButton();
});

Then("the user is successfully logged in and redirected to the dashboard", () => {
  cy.url().should("include", "/dashboard");
});

Given("the user is logged in", () => {
  LoginPage.ensureLoggedIn();
});

When("the user logs out of the account", () => {
  LoginPage.logout();
});

Then("the user is redirected to the login page", () => {
  cy.url().should("include", "/login");
});

When("the user selects 'Login with Facebook'", () => {
  LoginPage.clickLoginWithFacebook();
});

Then("the Facebook login page is displayed", () => {
  cy.url().should("include", "facebook.com");
});

When("the user enters Facebook credentials and logs in", () => {
  LoginPage.enterFacebookCredentials();
});

Then("the user is successfully logged in through Facebook and redirected to the dashboard", () => {
  cy.url().should("include", "/dashboard");
});

Given("the user is logged in through Facebook", () => {
  LoginPage.ensureLoggedInThroughFacebook();
});

When("the user selects 'Login with Google'", () => {
  LoginPage.clickLoginWithGoogle();
});

Then("the Google login page is displayed", () => {
  cy.url().should("include", "accounts.google.com");
});

When("the user enters Google credentials and logs in", () => {
  LoginPage.enterGoogleCredentials();
});

Then("the user is successfully logged in through Google and redirected to the dashboard", () => {
  cy.url().should("include", "/dashboard");
});

When("the user attempts to log in", () => {
  LoginPage.attemptLogin();
});

Then("CAPTCHA is displayed and required for login", () => {
  LoginPage.verifyCaptchaDisplayed();
});

Then("no security warnings or alerts are displayed", () => {
  LoginPage.verifyNoSecurityWarnings();
});

Given("the user navigates to the login page from a new device", () => {
  LoginPage.visitFromNewDevice();
});

Then("verification via email or phone is required", () => {
  LoginPage.verifyEmailOrPhoneVerificationRequired();
});

When("the user is inactive for a period", () => {
  LoginPage.waitForInactivity();
});

Then("the session times out and requires re-login", () => {
  LoginPage.verifySessionTimeout();
});

Then("all login pages and redirects are secured with HTTPS", () => {
  LoginPage.verifyHttpsRedirection();
});

Given("the system is under maintenance", () => {
  LoginPage.setSystemMaintenance();
});

Then("the user receives a maintenance notification and cannot log in", () => {
  LoginPage.verifyMaintenanceNotification();
});